sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+     * The version of Storage Analytics to configure.
+     */
+    version: string;
+    /**
+     * Indicates whether all delete requests should be logged.
+     */
+    deleteProperty: boolean;
+    /**
+     * Indicates whether all read requests should be logged.
+     */
+    read: boolean;
+    /**
+     * Indicates whether all write requests should be logged.
+     */
+    write: boolean;
+    retentionPolicy: RetentionPolicy;
+}
+/**
+ * a summary of request statistics grouped by API in hour or minute aggregates for blobs
+ */
+export interface Metrics {
+    /**
+     * The version of Storage Analytics to configure.
+     */
+    version?: string;
+    /**
+     * Indicates whether metrics are enabled for the Blob service.
+     */
+    enabled: boolean;
+    /**
+     * Indicates whether metrics should generate summary statistics for called API operations.
+     */
+    includeAPIs?: boolean;
+    retentionPolicy?: RetentionPolicy;
+}
+/**
+ * An interface representing PageRange.
+ */
+export interface PageRange {
+    start: number;
+    end: number;
+}
+/**
+ * An interface representing ClearRange.
+ */
+export interface ClearRange {
+    start: number;
+    end: number;
+}
+/**
+ * the list of pages
+ */
+export interface PageList {
+    pageRange?: PageRange[];
+    clearRange?: ClearRange[];
+}
+/**
+ * signed identifier
+ */
+export interface SignedIdentifier {
+    /**
+     * a unique id
+     */
+    id: string;
+    accessPolicy: AccessPolicy;
+}
+/**
+ * The properties that enable an account to host a static website
+ */
+export interface StaticWebsite {
+    /**
+     * Indicates whether this account is hosting a static website
+     */
+    enabled: boolean;
+    /**
+     * The default name of the index page under each directory
+     */
+    indexDocument?: string;
+    /**
+     * The absolute path of the custom 404 page
+     */
+    errorDocument404Path?: string;
+}
+/**
+ * Storage Service Properties.
+ */
+export interface StorageServiceProperties {
+    logging?: Logging;
+    hourMetrics?: Metrics;
+    minuteMetrics?: Metrics;
+    /**
+     * The set of CORS rules.
+     */
+    cors?: CorsRule[];
+    /**
+     * The default version to use for requests to the Blob service if an incoming request's version
+     * is not specified. Possible values include version 2008-10-27 and all more recent versions
+     */
+    defaultServiceVersion?: string;
+    deleteRetentionPolicy?: RetentionPolicy;
+    staticWebsite?: StaticWebsite;
+}
+/**
+ * Stats for the storage service.
+ */
+export interface StorageServiceStats {
+    geoReplication?: GeoReplication;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface LeaseAccessConditions {
+    /**
+     * If specified, the operation only succeeds if the resource's lease is active and matches this
+     * ID.
+     */
+    leaseId?: string;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface ModifiedAccessConditions {
```

Comment:
We recently discussed renaming this and putting it in Azure Core

Reply:
Looks good. The thing is will generated code depends on Azure Core?

Reply:
Can someone point me to this discussion?

Reply:
https://github.com/Azure/azure-sdk/issues/638